src/getMigrateDashboards.ts

```diff
@@ -41,25 +41,26 @@ export const getMigrateDashboards =
     },
   ) =>
   <FromDashboardState, ToDashboardState>(
+    deserialize: (state: any) => FromDashboardState,
```

Comment:
To avoid using `any`, we could have more type parameters, right?
Add `SerializedFromDashboardState` and `SerializedToDashboardState`, and modifying `FromDashboardState` --> `DeserializedFromDashboardState` and `ToDashboardState` --> `DeserializedToDashboardState`

Reply:
We could, but I decided against it for the following reasons:
- It would double the amount of generics without adding more type safety, as ultimately `any` is used anyway, and we're doing mutations.
- Having `FromSerializedDashboardState` and `FromDashboardState` as two different generics would not be aligned with `DashboardState<"serialized">` and `DashboardState`, where the serialization is differentiated with a generic in one same type, as opposed to having two different types.

Imo the `MigrationFunction` types there are helpful to give a structure and an outline to make the code easier to read, but not to ensure a 100% type safety.

Reply:
> Having FromSerializedDashboardState and FromDashboardState as two different generics would not be aligned with DashboardState<"serialized"> and DashboardState

I don't see that as a concern we should have. To me `@activeviam/activeui-sdk` is a different entity as `activeui-migration`, and we don't necessarily have in `activeui-migration` to follow the types structure that's defined in `@activeviam/activeui-sdk`. Especially if we need a different structure for a good reason.

>  imo the MigrationFunction types are mostly helpful to give a structure and an outline to make the code easier to read, but we cannot ensure a 100% type safety anyway.

> It would double the amount of generics without adding more type safety

I completely agree with those statements though. 

I'm leaning towards introducing the `FromSerializedXyzState` and `ToSerializedXyzState` types. I really think it would make the code easier to read, the currying structure with the generic serialization/deserialization making it pretty abstract. Especially with the order of the parameters that you went for, it would make the three steps (`deserialize`, `callback`, `serialize`) really clear when reading the function signature. 

It does double the amount of state types we use for those generic functions, but I think 4 type generics for the `getMigrateXyz` is still legible. It makes the ones of `MigrationFunction` go up to 12, which is a lot, but I noticed when used, `MigrationFunction` shouldn't need any type parameters, so I'm not concerned about it getting too heavy.

Are you happy to go for it?

On the same topic, I have an extra comment. I noticed we use `any` as the second type parameter of `callback` in [the signature of `migrateWidgetsWithinDashboard`](https://github.com/activeviam/activeui-migration/blob/main/src/migrateWidgetsWithinDashboard.ts#L23). I think we can introduce a `ToDashboardState` generic here too, to avoid using `any`, which makes the code less clear IMO.

What do you think?

Reply:
I pushed some commits with these changes: as expected it's more verbose, but I don't mind keeping it this way if you think it adds readability.
We could also keep just part of it: for example imo adding the generics to `migrate_5.0_to_5.1` does not add readability not more type safety than what we already had, so we could remove that part.

Reply:
Yeah, I agree we could not add the generics in `migrate_50_to_51`, at least for now. It would enlighten the function definition, and if we ever notice that it's better to have them, for whatever reason, we can add them back very quickly

But honestly I don't think I feel strongly about either options in this specific case, so as you already added them, feel free to leave this as it is.